client/src/pages/Catalog.js

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState, useCallback, useContext, Fragment } from 'react';
+import { fetchCatalog } from '../lib';
+import Card from "../components/Card";
+import Search from "../components/Search";
+import ToggleLamberto from '../components/ToggleLamberto';
+import { ShopContext } from '../components/ShopContext';
+
+export default function Catalog() {
+  const [inventory, setInventory] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+  const [searchResults, setsearchResults] = useState([]);
+  const { user } = useContext(ShopContext);
+
+  useEffect(() => {
+    async function loadCatalog() {
+      try {
+        const inventory = await fetchCatalog();
+        setInventory(inventory);
+      } catch (err) {
+        setError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    setIsLoading(true);
+    loadCatalog();
+  }, []);
+
+  const handleSearch = useCallback((results) => {
+    setsearchResults(results);
+  }, [])
+
+  if (isLoading) return (
+    <div className="d-flex justify-content-center align-items-center vh-100">
+      <div className="spinner-border text-primary" role="status">
+        <span className="sr-only visually-hidden">Loading...</span>
+      </div>
+    </div>
+  );
+
+  if (error) return (
+    <>
```

Comment:
I've seen this type of display several times. You probably should create a component to eliminate the code duplication.